keydata.go

```diff
@@ -61,14 +67,134 @@ type keyPolicyUpdateData struct {
 	CreationTicket *tpm2.TkCreation
 }
 
+type tpmObject struct {
+	Private tpm2.Private
+	Public  *tpm2.Public
+}
+
+type afSplitData struct {
+	Stripes uint32
+	HashAlg tpm2.HashAlgorithmId
+	Data    []byte
+}
+
+func (d *afSplitData) Marshal(w io.Writer) (nbytes int, err error) {
+	n, err := tpm2.MarshalToWriter(w, d.Stripes, d.HashAlg)
+	nbytes += n
+	if err != nil {
+		return nbytes, err
+	}
+
+	if err := binary.Write(w, binary.BigEndian, uint32(len(d.Data))); err != nil {
+		return nbytes, err
+	}
+	nbytes += binary.Size(uint32(0))
+
+	n, err = w.Write(d.Data)
+	nbytes += n
+	if err != nil {
+		return nbytes, err
+	}
+	return
+}
+
+func (d *afSplitData) Unmarshal(r io.Reader) (nbytes int, err error) {
+	n, err := tpm2.UnmarshalFromReader(r, &d.Stripes, &d.HashAlg)
+	nbytes += n
+	if err != nil {
+		return nbytes, err
+	}
+
+	var s uint32
+	if err := binary.Read(r, binary.BigEndian, &s); err != nil {
+		return nbytes, err
+	}
+	nbytes += binary.Size(s)
+
+	d.Data = make([]byte, s)
```

Comment:
I'm concerned about these values being used to allocated memory before copying from the reader -- I think we ought to bound these to reasonable limits.